### PR TITLE
update ecl and add param for innovation consistency checks for range …

### PR DESCRIPTION
**Pull request by ChristophTobler**
…aid fusion

includes https://github.com/PX4/ecl/pull/295

### CODE REVIEW

File: src/modules/ekf2/ekf2_main.cpp [hunk 1]
```diff
@@ -269,6 +269,8 @@ class Ekf2 : public control::SuperBlock
 	_rng_aid;              // enables use of a range finder even if primary height source is not range finder (EKF2_HGT_MODE != 2)
 	control::BlockParamExtFloat _rng_aid_hor_vel_max; 	// maximum allowed horizontal velocity for range aid
 	control::BlockParamExtFloat _rng_aid_height_max; 	// maximum allowed absolute altitude (AGL) for range aid
+	control::BlockParamExtFloat
+	_rng_aid_innov_gate; // gate size used for innovation consistency checks for range aid fusion
```
Comment (LorenzMeier) [inline src/modules/ekf2/ekf2_main.cpp]: Please use ```///<``` in the future instead. @priseborough - this tells Doxygen that this is a description of this class member.

Comment (priseborough) [inline src/modules/ekf2/ekf2_main.cpp]: OK, will do.

**Comment by ChristophTobler** [comment]
I rebased and updated ekf2_main to use `get_terrain_valid()` separately (as this changed in ecl).

**Comment by ChristophTobler** [comment]
includes https://github.com/PX4/ecl/pull/303#issuecomment-316302410

**Comment by priseborough** [comment]
@ChristophTobler  Please resolve conflicts.

File: src/modules/ekf2/ekf2_main.cpp (hunk 1 above)
Comment (priseborough) [inline src/modules/ekf2/ekf2_main.cpp]: I will go through the estimator code and submit a separate PR' for this and the ecl library to make the Doxygen compatibility changes.

**Comment by ChristophTobler** [comment]
@priseborough thanks!

**Pull request merged**